extension/src/popup/ducks/cache.ts

```diff
@@ -47,7 +51,10 @@ const cacheSlice = createSlice({
     saveBalancesForAccount(state, action: { payload: SaveBalancesPayload }) {
       state.balanceData = {
         ...state.balanceData,
-        [action.payload.publicKey]: action.payload.balances,
+        [action.payload.networkDetails.network]: {
```

Comment:
balances should be organized by network so we show the correct balance when a user switches network